src/crypto/key_expansion.rs

```diff
@@ -5,22 +5,58 @@ use super::{cipher_suite::CipherSuite, secret::Secret};
 use crate::error::Result;
 
 pub trait KeyExpansion {
-    fn expand_from<T>(cipher_suite: &CipherSuite, key_material: T) -> Result<Secret>
+    fn expand_from<M, K>(cipher_suite: &CipherSuite, key_material: M, key_id: K) -> Result<Secret>
     where
-        T: AsRef<[u8]>;
+        M: AsRef<[u8]>,
+        K: Into<u64>;
 }
 
-pub const SFRAME_HKDF_SALT: &[u8] = b"SFrame10";
-pub const SFRAME_HKDF_KEY_EXPAND_INFO: &[u8] = b"key";
-pub const SFRAME_HDKF_SALT_EXPAND_INFO: &[u8] = b"salt";
+pub fn get_hkdf_key_expand_info(key_id: u64) -> Vec<u8> {
+    [
+        SFRAME_LABEL,
+        SFRAME_HKDF_KEY_EXPAND_INFO,
+        &key_id.to_be_bytes(),
```

Comment:
is this length constant?

Reply:
It is actually.  We need to allocate here however (and can't return an iterator),  since openssl expects a slice for:https://docs.rs/openssl/latest/openssl/pkey_ctx/struct.PkeyCtx.html#method.set_hkdf_salt.

Creating an array on the stack makes the function quite complex:
```
    if #[cfg(feature = "openssl")] {
        pub const SFRAME_HKDF_SUB_ENC_EXPAND_INFO: &[u8] = b"enc";
        pub const SFRAME_HDKF_SUB_AUTH_EXPAND_INFO: &[u8] = b"auth";

        pub fn get_hkdf_aead_label(tag_len: usize) -> [u8; AEAD_LABEL_SIZE] {
            let mut label = [0; AEAD_LABEL_SIZE];
            label[..SFRAME_HDKF_SUB_AEAD_LABEL.len()].copy_from_slice(SFRAME_HDKF_SUB_AEAD_LABEL);
            // for current platforms there is no issue casting from usize to u64
            label[SFRAME_HDKF_SUB_AEAD_LABEL.len()..].copy_from_slice(&(tag_len as u64).to_be_bytes());
            label

        }
        const SFRAME_HDKF_SUB_AEAD_LABEL: &[u8] = b"SFrame 1.0 AES CTR AEAD ";
        const U64_BYTES:usize = (u64::BITS as usize)/8;
        const AEAD_LABEL_SIZE : usize = SFRAME_HDKF_SUB_AEAD_LABEL.len() + U64_BYTES;
    }
```
At least that was the best I could come up with. 
I think the extra complexity is not really worth it, since the key derivation is not performance critical, so I would rather keep it as is.